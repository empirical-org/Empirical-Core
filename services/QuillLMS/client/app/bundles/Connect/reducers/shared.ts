
export function receiveQuestionsData(currentstate, action) {
  if (action.language) {
    return Object.assign({}, currentstate, {
      translated_data: {
        ...currentstate.translated_data,
        [action.language] : action.data
      },
    });
  } else {
    return Object.assign({}, currentstate, {
      hasreceiveddata: true,
      data: action.data,
    });
  }
}

export function receiveQuestionData(currentstate, action) {
  return Object.assign({}, currentstate, {
    data: Object.assign({}, currentstate.data, {
      [action.uid]: action.data,
    })
  })
}
