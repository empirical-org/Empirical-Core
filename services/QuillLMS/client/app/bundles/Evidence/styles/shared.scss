.evidence-step-number-small {
  width: 100%;
  border-radius: 16px;
  justify-content: center;
  align-items: center;
  display: flex;
  font-size: 16px;
  font-weight: 600;
  margin-right: 8px;
  background-color: #005341;
  color: #ffffff;
  max-width: 24px;
  height: 24px;
  font-size: 14px;
  &.active {
    background-color: #ffffff;
    color: #262626;
  }
}
