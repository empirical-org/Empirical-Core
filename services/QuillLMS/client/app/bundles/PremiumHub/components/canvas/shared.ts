export interface CanvasInstance {
  client_id: string;
  id: number;
  client_secret: string;
  school_ids: number[];
  school_names: string[];
  url: string;
}

export interface School {
  id: number;
  name: string;
  subscription_status: any;
  subscriptions: any[];
}
