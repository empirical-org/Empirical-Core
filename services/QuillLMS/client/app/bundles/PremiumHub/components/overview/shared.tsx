export const USAGE_HIGHLIGHTS = 'Usage Highlights'
export const PROFESSIONAL_DEVELOPMENT_AND_SUPPORT = 'Professional Development and Support'
export const ACCOUNT_MANAGEMENT = 'Account Management'
export const PREMIUM_REPORTS = 'Premium Reports'
export const INTEGRATIONS = 'Integrations'

export const iconLinkBase = `${process.env.CDN_URL}/images/pages/administrator/overview`

export const SECTION_NAME_TO_ICON_URL = {
  [USAGE_HIGHLIGHTS]: `${iconLinkBase}/bulb.svg`,
  [PROFESSIONAL_DEVELOPMENT_AND_SUPPORT]: `${iconLinkBase}/students.svg`,
  [ACCOUNT_MANAGEMENT]: `${iconLinkBase}/pencil.svg`,
  [PREMIUM_REPORTS]: `${iconLinkBase}/bar-graph-increasing-black.svg`,
  [INTEGRATIONS]: `${iconLinkBase}/checkbox-multiple.svg`
}
