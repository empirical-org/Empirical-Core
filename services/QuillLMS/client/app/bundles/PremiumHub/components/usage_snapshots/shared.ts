export const SMALL = 'small'
export const MEDIUM = 'medium'
export const POSITIVE = 'positive'
export const NEGATIVE = 'negative'
