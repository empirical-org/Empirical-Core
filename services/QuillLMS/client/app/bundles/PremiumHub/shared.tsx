import * as React from 'react'
import * as _ from 'lodash'

import * as md5 from 'md5'

import { unorderedArraysAreEqual, } from '../../modules/unorderedArraysAreEqual'

export const RESTRICTED = 'restricted'
export const LIMITED = 'limited'
export const FULL = 'full'
export const LOADING = 'loading'

export const APPROVED = 'Approved'
export const PENDING = 'Pending'
export const DENIED = 'Denied'
export const SKIPPED = 'Skipped'

export const OVERVIEW = 'overview'
export const SKILL = 'skill'
export const STUDENT = 'student'

export const DIAGNOSTIC_REPORT_DEFAULT_CELL_WIDTH = '182px'
export const groupByDropdownOptions = [{ label: 'Grade', value: 'grade' }, { label: 'Teacher', value: 'teacher' }, { label: 'Classroom', value: 'classroom' }]
export const diagnosticTypeDropdownOptions = [
  { label: 'Starter Baseline Diagnostic (Pre)', value: 'Starter Baseline Diagnostic (Pre)' },
  { label: 'Starter Growth Diagnostic (Post)', value: 'Starter Growth Diagnostic (Post)' },
  { label: 'Intermediate Baseline Diagnostic (Pre)', value: 'Intermediate Baseline Diagnostic (Pre)' },
  { label: 'Intermediate Growth Diagnostic (Post)', value: 'Intermediate Growth Diagnostic (Post)' },
  { label: 'Advanced Baseline Diagnostic (Pre)', value: 'Advanced Baseline Diagnostic (Pre)' },
  { label: 'Advanced Growth Diagnostic (Post)', value: 'Advanced Growth Diagnostic (Post)' },
  { label: 'ELL Starter Baseline Diagnostic (Pre)', value: 'ELL Starter Baseline Diagnostic (Pre)' },
  { label: 'ELL Starter Growth Diagnostic (Post)', value: 'ELL Starter Growth Diagnostic (Post)' },
  { label: 'ELL Intermediate Baseline Diagnostic (Pre)', value: 'ELL Intermediate Baseline Diagnostic (Pre)' },
  { label: 'ELL Intermediate Growth Diagnostic (Post)', value: 'ELL Intermediate Growth Diagnostic (Post)' },
  { label: 'ELL Advanced Baseline Diagnostic (Pre)', value: 'ELL Advanced Baseline Diagnostic (Pre)' },
  { label: 'ELL Advanced Growth Diagnostic (Post)', value: 'ELL Advanced Growth Diagnostic (Post)' },
  { label: 'AP Writing Skills Survey', value: 'AP Writing Skills Survey' },
  { label: 'Pre-AP Writing Skills Survey 1', value: 'Pre-AP Writing Skills Survey 1' },
  { label: 'Pre-AP Writing Skills Survey 2', value: 'Pre-AP Writing Skills Survey 2' },
  { label: 'SpringBoard Writing Skills Survey', value: 'SpringBoard Writing Skills Survey' }
]

export const premiumLockImage = <img alt="Gray lock" src={`${process.env.CDN_URL}/images/pages/administrator/premium_lock.svg`} />

export const restrictedElement = (
  <div className="restricted">
    {premiumLockImage}
  </div>
)

export const restrictedPage = (
  <div className="restricted-page">
    {premiumLockImage}
  </div>
)

export function selectionsEqual(firstSelection, secondSelection) {
  return (
    firstSelection == secondSelection || // less strict comparison so that undefined and null are treated as equal
    unorderedArraysAreEqual(firstSelection, secondSelection)
  )
}

export function mapItemsIfNotAll(selectedItems, allItems, mapKey = 'id') {
  // selectedItems may, by design, be a superset of allItems, but if everything in allItems is in selectedItems, we want to treat it as "everything" being selected
  const allItemsSelected = allItems.every((item) => {
    return _.some(selectedItems, item)
  })

  if (allItemsSelected || selectionsEqual(selectedItems, allItems)) return null

  return selectedItems.map(i => i[mapKey])
}

/* Should produce the same output as backend `app/services/payload_hasher.rb:run` */
export function hashPayload(payloadArray: Array<any>) {
  const joinedPayload = payloadArray.join('-')
  return md5(joinedPayload)
}
