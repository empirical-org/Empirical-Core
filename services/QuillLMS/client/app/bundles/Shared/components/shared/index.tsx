export {
  ArchivedButton
} from './archivedButton'

export {
  PostNavigationBanner
} from './postNavigationBanner'

export {
  ButtonLoadingSpinner,
  DarkButtonLoadingSpinner,
  LightButtonLoadingSpinner,
} from './buttonLoadingSpinner'

export {
  Card
} from './card'

export {
  Checkbox
} from './checkBox'

export {
  DragHandle
} from './dragHandle'

export {
  ExpandableCard
} from './expandableCard'

export {
  CarouselAnimation
} from './carouselAnimation'

export {
  DataTable
} from './dataTable'

export {
  ProgressBar
} from './progressBar'

export {
  DropdownInput
} from './dropdownInput'

export {
  DropdownInputWithSearchTokens
} from './dropdownInputWithSearchTokens'

export {
  Error
} from './error'

export {
  FlagDropdown
} from './flagDropdown'

export {
  HelpfulTips
} from './helpfulTips'

export {
  Input
} from './input'

export {
  Passthrough
} from './passthrough'

export {
  ResumeOrBeginButton
} from './resumeOrBeginButton'

export {
  ReportHeader
} from './reportHeader'

export {
  ScreenreaderInstructions
} from './screenreaderInstructions'

export {
  OneThumbSlider,
  TwoThumbSlider,
} from './slider'

export {
  SmartSpinner
} from './smartSpinner'

export {
  Snackbar,
  defaultSnackbarTimeout
} from './snackbar'

export {
  SortableList
} from './sortableList'

export {
  Spinner
} from './spinner'

export {
  TeacherPreviewMenu
} from './teacherPreviewMenu'

export {
  TeacherPreviewMenuButton
} from './teacherPreviewMenuButton'

export {
  TextEditor
} from './textEditor'

export {
  TextArea
} from './textArea'

export {
  Tooltip
} from './tooltip'

export {
  UploadOptimalResponses
} from './uploadOptimalResponses'

export {
  ToggleComponentSection
} from './toggleComponentSection'
