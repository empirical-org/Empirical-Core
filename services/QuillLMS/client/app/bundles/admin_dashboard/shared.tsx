import * as React from 'react'

export const RESTRICTED = 'restricted'
export const LIMITED = 'limited'
export const FULL = 'full'

export const premiumLockImage = <img alt="Gray lock" src={`${process.env.CDN_URL}/images/pages/administrator/premium_lock.svg`} />

export const restrictedElement = (
  <div className="restricted">
    {premiumLockImage}
  </div>
)
