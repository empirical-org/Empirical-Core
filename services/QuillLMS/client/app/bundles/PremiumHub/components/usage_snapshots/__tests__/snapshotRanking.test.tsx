import { mount } from 'enzyme';
import * as React from 'react';

import { timeframes, grades, schools, teachers, classrooms, } from './data'

import { SMALL, NEGATIVE, POSITIVE, MEDIUM, } from '../shared'
import SnapshotRanking from '../snapshotRanking';

const data = [
  {value: "Comma Before Coordinating Conjunctions", count: 3777 },
  {value: "Compound Objects", count: 3339 },
  {value: "Subordinating Conjunction at the Beginning of a Sentence", count: 2232 },
  {value: "Gerunds as Subjects", count: 456 },
  {value: "Split Infinitives", count: 789 },
  {value: "Parallel Structure", count: 1234 },
  {value: "Subject-Verb Agreement", count: 5678 },
  {value: "Pronoun-Antecedent Agreement", count: 9012 },
  {value: "Appositive Phrases", count: 3456 },
  {value: "Dangling Modifiers", count: 7890 }
]


describe('SnapshotRanking component', () => {
  const mockProps = {
    label: 'Sentences written',
    queryKey: 'sentences-written',
    searchCount: 1,
    selectedGrades: grades.map(g => g.value),
    selectedSchoolIds: schools.map(s => s.id),
    selectedTeacherIds: teachers.map(s => s.id),
    selectedClassroomIds: classrooms.map(s => s.id),
    selectedTimeframe: timeframes[0].value,
    adminId: 1,
    customTimeframeStart: null,
    customTimeframeEnd: null,
    comingSoon: false,
    headers: ['Teacher', 'Activities completed']
  }

  describe ('when it is coming soon', () => {
    it('should match snapshot', () => {
      const component = mount(<SnapshotRanking {...mockProps} comingSoon={true} />);

      expect(component).toMatchSnapshot();
    });
  })

  describe ('when there is no data ', () => {
    it('should match snapshot', () => {
      const component = mount(<SnapshotRanking {...mockProps} />);

      expect(component).toMatchSnapshot();
    });
  })

  describe ('when there is data ', () => {
    it('should match snapshot', () => {
      const component = mount(<SnapshotRanking {...mockProps} passedData={data} />);

      expect(component).toMatchSnapshot();
    });
  })


});
