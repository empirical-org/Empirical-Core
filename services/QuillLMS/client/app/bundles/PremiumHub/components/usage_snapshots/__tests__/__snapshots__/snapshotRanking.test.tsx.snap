// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SnapshotRanking component when it is coming soon should match snapshot 1`] = `
<SnapshotRanking
  adminId={1}
  comingSoon={true}
  customTimeframeEnd={null}
  customTimeframeStart={null}
  headers={
    Array [
      "Teacher",
      "Activities completed",
    ]
  }
  label="Sentences written"
  queryKey="sentences-written"
  searchCount={1}
  selectedClassroomIds={
    Array [
      170171,
      1203848,
      1203849,
    ]
  }
  selectedGrades={
    Array [
      "Kindergarten",
      "1",
      "2",
      "3",
      "4",
      "5",
      "6",
      "7",
      "8",
      "9",
      "10",
      "11",
      "12",
      "University",
      "Other",
    ]
  }
  selectedSchoolIds={
    Array [
      58452,
    ]
  }
  selectedTeacherIds={
    Array [
      1500140,
      1500832,
      3570928,
    ]
  }
  selectedTimeframe="last-30-days"
>
  <section
    className="snapshot-item snapshot-ranking no-data"
  >
    <div
      onClick={[Function]}
    >
      <div
        className="header"
      >
        <h3>
          Sentences written
        </h3>
      </div>
      <DataTable
        data={null}
        headers={
          Array [
            "Teacher",
            "Activities completed",
          ]
        }
        numberOfRows={3}
      >
        <table
          className="table"
        >
          <tr
            className="header-row"
          >
            <th
              key="Teacher"
            >
              Teacher
            </th>
            <th
              key="Activities completed"
            >
              Activities completed
            </th>
          </tr>
          <tr
            className="data-row"
            key="0"
          >
            <td>
              —
            </td>
            <td>
              —
            </td>
          </tr>
          <tr
            className="data-row"
            key="1"
          >
            <td>
              —
            </td>
            <td>
              —
            </td>
          </tr>
          <tr
            className="data-row"
            key="2"
          >
            <td>
              —
            </td>
            <td>
              —
            </td>
          </tr>
        </table>
      </DataTable>
    </div>
  </section>
</SnapshotRanking>
`;

exports[`SnapshotRanking component when there is data  should match snapshot 1`] = `
<SnapshotRanking
  adminId={1}
  comingSoon={false}
  customTimeframeEnd={null}
  customTimeframeStart={null}
  headers={
    Array [
      "Teacher",
      "Activities completed",
    ]
  }
  label="Sentences written"
  passedData={
    Array [
      Object {
        "count": 3777,
        "value": "Comma Before Coordinating Conjunctions",
      },
      Object {
        "count": 3339,
        "value": "Compound Objects",
      },
      Object {
        "count": 2232,
        "value": "Subordinating Conjunction at the Beginning of a Sentence",
      },
      Object {
        "count": 456,
        "value": "Gerunds as Subjects",
      },
      Object {
        "count": 789,
        "value": "Split Infinitives",
      },
      Object {
        "count": 1234,
        "value": "Parallel Structure",
      },
      Object {
        "count": 5678,
        "value": "Subject-Verb Agreement",
      },
      Object {
        "count": 9012,
        "value": "Pronoun-Antecedent Agreement",
      },
      Object {
        "count": 3456,
        "value": "Appositive Phrases",
      },
      Object {
        "count": 7890,
        "value": "Dangling Modifiers",
      },
    ]
  }
  queryKey="sentences-written"
  searchCount={1}
  selectedClassroomIds={
    Array [
      170171,
      1203848,
      1203849,
    ]
  }
  selectedGrades={
    Array [
      "Kindergarten",
      "1",
      "2",
      "3",
      "4",
      "5",
      "6",
      "7",
      "8",
      "9",
      "10",
      "11",
      "12",
      "University",
      "Other",
    ]
  }
  selectedSchoolIds={
    Array [
      58452,
    ]
  }
  selectedTeacherIds={
    Array [
      1500140,
      1500832,
      3570928,
    ]
  }
  selectedTimeframe="last-30-days"
>
  <section
    className="snapshot-item snapshot-ranking has-data"
  >
    <div
      onClick={[Function]}
    >
      <div
        className="header"
      >
        <h3>
          Sentences written
        </h3>
        <button
          aria-label="Open modal with additional seven lines of table data"
          className="interactive-wrapper focus-on-light"
          onClick={[Function]}
          type="button"
        >
          <img
            alt=""
            src="undefined/images/pages/administrator/expand.svg"
          />
        </button>
      </div>
      <DataTable
        data={
          Array [
            Object {
              "count": 3777,
              "value": "Comma Before Coordinating Conjunctions",
            },
            Object {
              "count": 3339,
              "value": "Compound Objects",
            },
            Object {
              "count": 2232,
              "value": "Subordinating Conjunction at the Beginning of a Sentence",
            },
            Object {
              "count": 456,
              "value": "Gerunds as Subjects",
            },
            Object {
              "count": 789,
              "value": "Split Infinitives",
            },
            Object {
              "count": 1234,
              "value": "Parallel Structure",
            },
            Object {
              "count": 5678,
              "value": "Subject-Verb Agreement",
            },
            Object {
              "count": 9012,
              "value": "Pronoun-Antecedent Agreement",
            },
            Object {
              "count": 3456,
              "value": "Appositive Phrases",
            },
            Object {
              "count": 7890,
              "value": "Dangling Modifiers",
            },
          ]
        }
        headers={
          Array [
            "Teacher",
            "Activities completed",
          ]
        }
        numberOfRows={3}
      >
        <table
          className="table"
        >
          <tr
            className="header-row"
          >
            <th
              key="Teacher"
            >
              Teacher
            </th>
            <th
              key="Activities completed"
            >
              Activities completed
            </th>
          </tr>
          <tr
            className="data-row"
            key="0"
          >
            <td>
              Comma Before Coordinating Conjunctions
            </td>
            <td>
              3777
            </td>
          </tr>
          <tr
            className="data-row"
            key="1"
          >
            <td>
              Compound Objects
            </td>
            <td>
              3339
            </td>
          </tr>
          <tr
            className="data-row"
            key="2"
          >
            <td>
              Subordinating Conjunction at the Beginning of a Sentence
            </td>
            <td>
              2232
            </td>
          </tr>
        </table>
      </DataTable>
    </div>
  </section>
</SnapshotRanking>
`;

exports[`SnapshotRanking component when there is no data  should match snapshot 1`] = `
<SnapshotRanking
  adminId={1}
  comingSoon={false}
  customTimeframeEnd={null}
  customTimeframeStart={null}
  headers={
    Array [
      "Teacher",
      "Activities completed",
    ]
  }
  label="Sentences written"
  queryKey="sentences-written"
  searchCount={1}
  selectedClassroomIds={
    Array [
      170171,
      1203848,
      1203849,
    ]
  }
  selectedGrades={
    Array [
      "Kindergarten",
      "1",
      "2",
      "3",
      "4",
      "5",
      "6",
      "7",
      "8",
      "9",
      "10",
      "11",
      "12",
      "University",
      "Other",
    ]
  }
  selectedSchoolIds={
    Array [
      58452,
    ]
  }
  selectedTeacherIds={
    Array [
      1500140,
      1500832,
      3570928,
    ]
  }
  selectedTimeframe="last-30-days"
>
  <section
    className="snapshot-item snapshot-ranking no-data"
  >
    <div
      onClick={[Function]}
    >
      <div
        className="header"
      >
        <h3>
          Sentences written
        </h3>
      </div>
      <DataTable
        data={null}
        headers={
          Array [
            "Teacher",
            "Activities completed",
          ]
        }
        numberOfRows={3}
      >
        <table
          className="table"
        >
          <tr
            className="header-row"
          >
            <th
              key="Teacher"
            >
              Teacher
            </th>
            <th
              key="Activities completed"
            >
              Activities completed
            </th>
          </tr>
          <tr
            className="data-row"
            key="0"
          >
            <td>
              —
            </td>
            <td>
              —
            </td>
          </tr>
          <tr
            className="data-row"
            key="1"
          >
            <td>
              —
            </td>
            <td>
              —
            </td>
          </tr>
          <tr
            className="data-row"
            key="2"
          >
            <td>
              —
            </td>
            <td>
              —
            </td>
          </tr>
        </table>
      </DataTable>
    </div>
  </section>
</SnapshotRanking>
`;
